### PR TITLE
docker + fig based development environment setup to run celery.

### PR DESCRIPTION
**Pull request by bhanuvrat**
added bare minimum configurations required to run celery.
and threw in docker and fig to automate setting up the development environment.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM ubuntu:14.04
+MAINTAINER Anuvrat Parashar "anuvrat@anuvrat.in"
+
+RUN apt-get update && apt-get -y upgrade && apt-get install -y git python2.7 python-pip python-dev postgresql-server-dev-all
+ADD requirements.txt /srv/requirements.txt
```
Comment (theskumar) [inline Dockerfile]: Can we use `requirements-dev.txt` here? something like this looks reasonable:

```
ADD requirements-dev.txt /srv/requirements-dev.txt
WORKDIR /srv/
RUN pip install -r /srv/requirements-dev.txt
```

**Comment by theskumar** [comment]
I tried this on mac with `boot2docker` and things look good. On mac, there are two difference, though.

1. You don't need to use `sudo`
2. You need to use `boot2docker ip` to obtain the IP of host of web server.

Wondering if this we can have it documented as well.

**Comment by theskumar** [comment]
Also can we have celery settings not only a dev settings but as a general setting.

in `settings/__init__.py`

```
from .celery import *  # noqa
```

and in `settings/celery.py`

```
import os
from kombu import Queue

from .common import TIME_ZONE as DJANGO_TIME_ZONE

BROKER_URL = os.environ.get("BROKER_URL", "redis://localhost:6379/0")
CELERY_RESULT_BACKEND = os.environ.get("CELERY_RESULT_BACKEND", 'redis://localhost:6379/0')
CELERY_ACCEPT_CONTENT = ['json']

CELERY_TIMEZONE = DJANGO_TIME_ZONE
CELERY_ENABLE_UTC = True

CELERY_DEFAULT_QUEUE = 'tasks'
CELERY_QUEUES = (
    Queue('tasks', routing_key='task.#'),
    Queue('transient', routing_key='transient.#', delivery_mode=1)
)
CELERY_DEFAULT_EXCHANGE = 'tasks'
CELERY_DEFAULT_EXCHANGE_TYPE = 'topic'
CELERY_DEFAULT_ROUTING_KEY = 'task.default'
```

**Comment by theskumar** [comment]
And one more things please add yourself to [CONTRIBUTORS.txt](https://github.com/pythonindia/junction/blob/master/CONTRIBUTORS.txt) :blush:

**Comment by theskumar** [comment]
It seems `CELERY_ACCEPT_CONTENT` is not enough. You need to add:

```
CELERY_TASK_SERIALIZER = 'json'
CELERY_RESULT_SERIALIZER = 'json'
```
Ref: http://stackoverflow.com/a/20928529/782901

I got the following error while trying a simple `def say_hello(): return 1+1` task

```
ContentDisallowed: Refusing to deserialize untrusted content of type pickle (application/x-python-serialize)
```

File: Dockerfile (hunk 1 above)
Comment (bhanuvrat) [inline Dockerfile]: could be done, sure.
btw, are there package mentioned in requirements-dev.txt that would be essential for running the project inside docker?

Comment (theskumar) [inline Dockerfile]: One of the major things that is required during development is testing libraries.

Comment (theskumar) [inline Dockerfile]: Would love to run the tests in the same production like environment.

**Comment by bhanuvrat** [comment]
assuming that `fig up` has already been fired up, test celery's integration by fire up a shell with

    fig run web /usr/bin/python manage.py shell

now in the shell, run the following two commands

    from junction.conferences import tasks
    tasks.run.delay(1,2)

the output should materialize on the logs printed by `fig`

File: Dockerfile (hunk 1 above)
Comment (bhanuvrat) [inline Dockerfile]: got it.

**Comment by theskumar** [comment]
The celery tasks run super smooth. Can we have `requirements-dev.txt` added to docker? I think we are good to close the PR after that :)

**Comment by theskumar** [comment]
This looks all good to me. Merging this in.

**Pull request merged**